core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +314,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput' &&
+                  maxSelectionCount === 1) {
+        var numChoices = $scope.getAnswerChoices().length;
+        choiceRules = [];
+        for (var i = 0; i < answerGroups.length; i++) {
+          choiceRules.push(false);
+        }
+
+        var newChoiceRules = function(current) {
+          return choiceRules.map(function(element, index) {
+            if (index !== current || element === true) {
+              return true;
+            } else if (index === current && element !== true) {
+              return false;
+            }
+          });
+        };
+
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              // If the rule_type of the first input is equalTo or containsOne,
+              // then set first choiceRules to true
+              if (ruleSpecs[j].rule_type === 'Equals') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type === 'ContainsAtLeastOneOf') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type ===
+                'DoesNotContainAtLeastOneOf') {
+                // If the ruleType is omits one of, then set the others to true
+                var current = i;
+                choiceRules = newChoiceRules(current);
```

Comment:
So let's say you have 3 answer choices A, B, C, and 5 answer groups of the form:

- Equal to A
- Does not contain at least one of [B]
- Equal to C
- Contains at least one of [B, C]
- Does not contain at least one of [A, B]

and you have an array for the three choices that starts off as [false, false, false].

After reading the first answer group, this becomes [true, false, false].
After the second, it becomes [true, false, true].
After the third, it's still [true, false, true].
After the fourth, it's [true, true, true].
After the fifth, it's [true, true, true] still.

Note that once an entry has been marked true, it remains true, since "true" means that the answer has been handled by at least one answer group. Also, in the above computation, there's no need for a choiceRules. Does that help?